### PR TITLE
feat(android): add platform-specific API for clearing package data

### PR DESCRIPTION
**Pull request by lokendrarawat97**
- [x] This is a small change 
- [ ] This change has been discussed in issue #<?> and the solution has been agreed upon with maintainers.

---

**Description:**

added clearStorage option to clear app data for android

### CODE REVIEW

**Comment by lokendrarawat97** [comment]
@noomorph ok will do it today.

File: detox/src/devices/Device.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ class Device {
   }
 
   async resetContentAndSettings() {
-    await this.deviceDriver.resetContentAndSettings(this._deviceId);
+    await this.deviceDriver.resetContentAndSettings(this._deviceId, this._bundleId);
```
Comment (noomorph) [inline detox/src/devices/Device.js]: @LeoNatan, I now wonder whether `resetContentAndSettings` process can be made app-specific, not global for the whole device (simulator)?

Comment (LeoNatan) [reply detox/src/devices/Device.js]: Not possible. On iOS there is no such functionality, and it is the same with simulators. We can delete the app, but that's not the same thing and doesn't reset the settings related to the app.

**Comment by heroic** [comment]
@noomorph 
`adb shell recovery --wipe_data` is not recommended or even useful, as it reboots the emulator. In essence `shell pm clear ${packageId}` resets the app, not the phone. There isn't a way to do a light factory reset on android without a reboot.

So it would probably be better to use it as a new method.

**Comment by noomorph** [comment]
@heroic, if you check the behavior on iOS, that's exactly what resetContentAndSettings does - wipes and reboots a device. I see no contradictions there.

As for clean app data, I agree and that is what I also said earlier, that it has to be added as a New method

**Pull request closed without merging**